internal/iso_packaging/iso_packaging.go

```diff
@@ -0,0 +1,241 @@
+)
+
+// Builder is a struct that manages a build to package an iso
+type Builder struct {
+	client      client.Client
+	jobSpec     *batchv1.Job
+	build       *v1alpha1.OSBuild
+	buildConfig *v1alpha1.OSBuildEnvConfig
+	baseImage   string
+}
+
+func NewBuilderJob(client client.Client, build *v1alpha1.OSBuild, buildConfig *v1alpha1.OSBuildEnvConfig, image string) (*Builder, error) {
+
+	if build == nil || buildConfig == nil {
+		return nil, fmt.Errorf("Invalid config added")
+	}
+
+	return &Builder{
+		client:      client,
+		jobSpec:     nil,
+		build:       build,
+		buildConfig: buildConfig,
+		baseImage:   image,
+	}, nil
+}
+
+func (b *Builder) Start() error {
+	sourceISO := b.build.Status.ComposerIso
+	if sourceISO == "" {
+		return fmt.Errorf("Cannot parse invalid iso image")
+	}
+	_, err := url.Parse(sourceISO)
+	if err != nil {
+		return err
+	}
+
+	if b.build.Spec.Kickstart == nil {
+		return fmt.Errorf("Kickstart is not defined, skipping")
+	}
+
+	command := fmt.Sprintf(templateCommand, b.build.Status.ComposerIso, kickstart)
+
+	jobSpec := &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      b.build.Name,
+			Namespace: b.build.Namespace,
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					APIVersion: b.build.APIVersion,
+					Kind:       b.build.Kind,
+					Name:       b.build.Name,
+					UID:        b.build.UID,
+				},
+			},
+		},
+		Spec: batchv1.JobSpec{
+			TTLSecondsAfterFinished: &jobTTLafterFinish,
+			ActiveDeadlineSeconds:   &deadlineSeconds,
+			Template: v1.PodTemplateSpec{
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{
+						{
+							Name:    "iso-command",
+							Image:   b.baseImage,
+							Command: strings.Split(command, " "),
+							VolumeMounts: []v1.VolumeMount{
+								{
+									Name:      "config",
+									ReadOnly:  true,
+									MountPath: "/opt/iso_package/",
+								}},
+						},
+					},
+					RestartPolicy: v1.RestartPolicyNever,
+					Volumes: []v1.Volume{
+						{
+							Name: "config",
+							VolumeSource: v1.VolumeSource{
+								Projected: &v1.ProjectedVolumeSource{
+									Sources: []v1.VolumeProjection{{
+										ConfigMap: &v1.ConfigMapProjection{
+											LocalObjectReference: v1.LocalObjectReference{
+												Name: b.build.Spec.Kickstart.Name,
+											},
+										},
+									}},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	if b.buildConfig.Spec.S3Service.AWS != nil {
+
+		config := b.buildConfig.Spec.S3Service.AWS
+
+		jobSpec.Spec.Template.Spec.Containers[0].EnvFrom = []v1.EnvFromSource{
+			{
+				SecretRef: &v1.SecretEnvSource{
+					LocalObjectReference: v1.LocalObjectReference{Name: config.CredsSecretReference.Name},
+					Optional:             aws.Bool(false)},
+			},
+		}
+		jobSpec.Spec.Template.Spec.Containers[0].Env = []v1.EnvVar{
+			{
+				Name:  "AWS_DEFAULT_REGION",
+				Value: config.Region,
+			},
+		}
+
+		jobSpec.Spec.Template.Spec.Containers[0].Command = append(
```

Comment:
The `append` call for the `Command` is the same for both cases

Reply:
Looks similar, but are different: 

Config is retrieved here: 
https://github.com/eloycoto/osbuild-operator/blob/55f89a026a4a4c7feb921daf342e6d30ed976b30/internal/iso_packaging/iso_packaging.go#L124

And here: 

https://github.com/eloycoto/osbuild-operator/blob/55f89a026a4a4c7feb921daf342e6d30ed976b30/internal/iso_packaging/iso_packaging.go#L146

Reply:
You're right, my bad